dashboard/src/components/toaster/toaster.tsx

```diff
@@ -0,0 +1,95 @@
+import { useAppSelector } from '@/hooks/redux.hook';
+import {
+    selectToaster,
+    toasterActions,
+    ToasterStatus,
+    ToasterType,
+} from '@/redux/reducers/toaster.reducer';
+import { useEffect, useRef, useState } from 'react';
+import { ToasterState } from '@/redux/reducers/toaster.reducer';
+import { useDispatch } from 'react-redux';
+import iconCloseSrc from '@/assets/images/light-close.svg';
+
+const Toaster = () => {
+    const toasters = useAppSelector(selectToaster).toasters;
+    const init: ToasterState[] = [];
+    const previousToastersRef = useRef(init);
```

Comment:
```suggestion 
    const previousToastersRef = useRef<ToasterState[>([]);
```

這樣一來應該不用多宣告一個 `init` 變數